test/cli/whole-program_test.py

```diff
@@ -358,3 +359,28 @@ def test_checkclass_project_builddir_j(tmpdir):
     build_dir = os.path.join(tmpdir, 'b1')
     os.mkdir(build_dir)
     __test_checkclass_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
+
+def __test_nullpointer_file0(extra_args):
+    args = [
+        '-q',
+        '--xml',
+        '--error-exitcode=1',
+        'whole-program/nullpointer1.cpp'
+    ]
+
+    args += extra_args
+
+    ret, stdout, stderr = cppcheck(args, cwd=__script_dir)
+    results = ET.fromstring(stderr)
+    file0 = ''
+    for e in results.findall('errors/error'):
+        if e.attrib['id'] == 'ctunullpointer':
+            if 'file0' in e.attrib:
```

Comment:
Not sure why this is necessary, but there was a `KeyError` without it.

Reply:
That means that it doesn't contain that key. `[]` acts like `at()`in C++.

Reply:
Yeah, but there should only be one `ctunullpointer` error, which has the `file0` attribute.
XML shown here: https://github.com/danmar/cppcheck/actions/runs/12605458944/job/35134122531
KeyError: https://github.com/danmar/cppcheck/actions/runs/12605258027/job/35133607619

Reply:
We should probably add a few more asserts on the results. Like the amount of returned nodes and such.

You could also limit the XPath to only return the node in question by adding `[@id='ctunullpointer']`.

Reply:
> We should probably add a few more asserts on the results. Like the amount of returned nodes and such.

Not so easy, since getting both `nullpointer` and `ctunullpointer` seems wrong already. With `-j2`, we get an additional `ctunullpointer` (https://trac.cppcheck.net/ticket/11883)
> You could also limit the XPath to only return the node in question by adding `[@id='ctunullpointer']`.

Done.